File: lib/agent/intent/coach_intent.dart
enum CoachIntentType {
  safety,
  generatePlan,
  compressWorkout,
  replaceExercise,
  rescheduleWeek,
  moveWorkoutSession,
  trainingFeedback,
  feedbackAdjustment,
  recoveryAdvice,
  nutritionAdvice,
  clarification,
  unrelated,
}

class IntentCandidate {
  const IntentCandidate({
    required this.type,
    required this.score,
    this.reason,
    this.slots = const {},
    this.missingSlots = const [],
  });

  final CoachIntentType type;
  final double score;
  final String? reason;
  final Map<String, dynamic> slots;
  final List<String> missingSlots;

  bool get hasMissingSlots => missingSlots.isNotEmpty;
}

File: agent_backend/agents/intent/coach_intent.py
from __future__ import annotations

from dataclasses import dataclass, field
from enum import Enum
from typing import Any


class CoachIntentType(str, Enum):
    safety = "safety"
    generatePlan = "generatePlan"
    compressWorkout = "compressWorkout"
    replaceExercise = "replaceExercise"
    rescheduleWeek = "rescheduleWeek"
    moveWorkoutSession = "moveWorkoutSession"
    trainingFeedback = "trainingFeedback"
    recoveryAdvice = "recoveryAdvice"
    nutritionAdvice = "nutritionAdvice"
    clarification = "clarification"
    unrelated = "unrelated"


@dataclass(frozen=True)
class IntentCandidate:
    type: CoachIntentType
    score: float
    reason: str | None = None
    slots: dict[str, Any] = field(default_factory=dict)
    missing_slots: list[str] = field(default_factory=list)

    @property
    def has_missing_slots(self) -> bool:
        return bool(self.missing_slots)
